.github/actions/build-docs/action.yml

```diff
@@ -0,0 +1,22 @@
+name: "Build client documentation"
+description: "Generates client documentation using pdoc"
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.x
+
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+
+    - name: Install dependencies
+      shell: bash
+      run: |
+        poetry install
+
+    - name: Build html documentation
+      shell: bash
+      run: |
+        poetry run pdoc pinecone/ --docformat google -o ./docs
```

Comment:
The `--docformat google` bit is here because it looked to me like a lot of the generated comments for the OpenAPI stuff, and our comments in general, were of this format. At least it seemed like the most reasonable visual output given existing comments. Info here:

https://pdoc.dev/docs/pdoc.html#use-numpydoc-or-google-docstrings